### PR TITLE
:sparkles: add new addonHealthCheck func to check all fields and support wildcard

### PR DESCRIPTION
**Pull request by zhiweiyin318**
## Summary
1. add new addonHealthCheck func to check all filed results 
2. support wildcard in the ProbeField

## Related issue(s)
https://github.com/open-cluster-management-io/addon-framework/issues/288
https://github.com/open-cluster-management-io/addon-framework/issues/287
Fixes #

### CODE REVIEW

File: pkg/agent/inteface.go [hunk 1]
```diff
@@ -169,13 +169,19 @@ type HealthProber struct {
 }
 
 type AddonHealthCheckFunc func(workapiv1.ResourceIdentifier, workapiv1.StatusFeedbackResult) error
+type AddonHealthCheckerFunc func([]ResultField) error
```
Comment (zhujian7) [inline pkg/agent/inteface.go]: Is the managedclusteraddon or managedcluster also required in some cases?
I remember someone raised a similar demand during our offline discussion?

File: pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go [hunk 2]
```diff
@@ -167,17 +168,38 @@ func (s *healthCheckSyncer) probeAddonStatusByWorks(
 		manifestConditions = append(manifestConditions, work.Status.ResourceStatus.Manifests...)
 	}
 
-	probeFields, healthChecker, err := s.analyzeWorkProber(s.agentAddon, cluster, addon)
+	probeFields, healthChecker, healthAllChecker, err := s.analyzeWorkProber(s.agentAddon, cluster, addon)
```
Comment (zhujian7) [inline pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go]: I don't think `healthAllChecker` is a proper name here. seems no meaning for `All`?

File: pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go [hunk 3]
```diff
@@ -167,17 +168,38 @@ func (s *healthCheckSyncer) probeAddonStatusByWorks(
...
 	if err != nil {
 		// should not happen, return
 		return err
 	}
 
+	var resultFields []agent.ResultField
+
 	for _, field := range probeFields {
```
Comment (zhujian7) [inline pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go:180]: can we separate the healthChecker(Deprecated one) and healthAllChecker logic into two different functions, so it is clear and easy to deprecate in the future?

File: pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go [hunk 4]
```diff
@@ -294,27 +331,46 @@ func (s *healthCheckSyncer) analyzeWorkloadsWorkProber(
...
 
-		if len(status.StatusFeedbacks.Values) == 0 {
-			return nil
-		}
+	return results
+}
 
-		return &status.StatusFeedbacks
+// compare two string, target may include *
+func wildcardMatch(resource, target string) bool {
```
Comment (zhujian7) [inline pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go:357]: These seem to be duplicates of the ocm/work code?

File: pkg/agent/inteface.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/agent/inteface.go]: @qiujian16  looks like it makes sense to input cluster and mca. WDYT?

File: pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go (hunk 3 above)
Comment (zhiweiyin318) [reply pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go:180]: Add a TODO here, will follow up to refactor this part

File: pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go (hunk 4 above)
Comment (zhiweiyin318) [reply pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go:357]: yes. only this two parts use it

File: pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/addonmanager/controllers/agentdeploy/healthcheck_sync.go]: re-named

**Pull request merged**